python/paddle/distributed/launch.py

```diff
@@ -212,12 +240,25 @@ def start_procs(args):
                 logger.warning("Use Cloud specified port:{}.".format(
                     cloud_paddle_port))
 
+    free_ports = None
+    if not args.use_paddlecloud and num_nodes <= 1:
+        free_ports = find_free_ports(selected_gpus_num)
+        if free_ports is not None:
+            free_ports = list(free_ports)
+
+    if free_ports is None:
+        free_ports = [
+            x
+            for x in range(args.started_port, args.started_port +
```

Comment:
What's the priority of the started_port config and the free_ports? It seems that if the free_ports are available, the launcher will use free_ports regardless of started_port set by user.

Reply:
Done.